plugins/inputs/http_response/http_response.go

```diff
@@ -148,6 +154,7 @@ func (h *HTTPResponse) httpGather() (map[string]interface{}, error) {
 
 	fields["response_time"] = time.Since(start).Seconds()
 	fields["http_response_code"] = resp.StatusCode
+	fields["connection_failed"] = 0
```

Comment:
One consequence of combining them is that we would not be able to store how long the timeout took, but maybe we don't care about this since it is basically set by hand.

Interesting idea to work match/mismatch in, do you think we would leave the field off if no match string is provided?  Perhaps we have a success result_type?

I was letting the int slide here because this plugin already has the response match field with one, but yeah in general we should use booleans.